database/auth.go

```diff
@@ -114,6 +114,55 @@ func (db *Db) storeVestingAccount(account exported.VestingAccount) (int, error)
 	return vestingAccountRowID, nil
 }
 
+func (db *Db) StoreBaseVestingAccountFromMsg(bva *vestingtypes.BaseVestingAccount) error {
+	stmt := `
+	INSERT INTO vesting_account (type, address, original_vesting, end_time) 
+	VALUES ($1, $2, $3, $4)
+	ON CONFLICT (address) DO UPDATE 
+		SET type = excluded.type,
+			original_vesting = excluded.original_vesting, 
+			end_time = excluded.end_time`
+
+	_, err := db.Sql.Exec(stmt,
+		proto.MessageName(bva),
+		bva.GetAddress().String(),
+		pq.Array(dbtypes.NewDbCoins(bva.OriginalVesting)),
+		time.Unix(bva.EndTime, 0))
+	if err != nil {
+		return fmt.Errorf("error while storing vesting account: %s", err)
+	}
+	return nil
+}
+
+func (db *Db) StorePeriodicVestingAccountFromMsg(pva *vestingtypes.PeriodicVestingAccount) error {
+	stmt := `
+	INSERT INTO vesting_account (type, address, start_time) 
```

Comment:
Can you also store `vesting_periods` info in db?

Reply:
hihi @MonikaCat , the vesting periods are stored in a different table "vesting_periods". It is handled at line 158 of this file.

Reply:
Ok 👍🏻